src/libstd/keyword_docs.rs

```diff
@@ -1167,12 +1185,31 @@ mod await_keyword {}
 
 #[doc(keyword = "dyn")]
 //
-/// Name the type of a [trait object].
+/// `dyn` is a prefix of a [trait object]'s type.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// The `dyn` keyword is used to highlight that calling methods on a trait is to be achieved by
+/// dynamic dispatch. An instance of a struct passed this way is called a 'trait object' and the
```

Comment:
```suggestion
/// dynamic dispatch. An instance of a struct passed this way is called a 'trait object'.
/// To use the trait this way, it must be 'object safe'.
```
(General note: "in order" increases wordiness in writing, is redundant, and should be removed.)